tests/test_bodc_queries.py

```diff
@@ -0,0 +1,50 @@
+# TODO the imports for clients should be able to be just on the toplevel:
+# from pykg2tbl import KGSource, QueryResult, DefaultQueryBuilder
+from const import BODC_ENDPOINT, FAKE_DUMP_FILE, P06_DUMP_FILE
+
+from pykg2tbl.j2.jinja_sparql_builder import (
+    J2SparqlBuilder as DefaultQueryBuilder,
+)
+from pykg2tbl.kg2tbl import KGSource
+from pykg2tbl.query import QueryResult
+
+
+def test_bodc_listing_published_P06():
+    nerc_server: KGSource = KGSource.build(BODC_ENDPOINT)
+    qry: str = DefaultQueryBuilder().build_sparql_query(
+        name="bodc-listing.sparql", cc="P06"
+    )
+
+    result: QueryResult = nerc_server.query(sparql=qry)
+    assert result is not None, "there should be a result"
+    assert len(result) > 0, "the result should not be empty"
+
+
+def test_bodc_listing_knowndump_P06():
+    ttl_dump = P06_DUMP_FILE
+    assert (
+        ttl_dump.exists()
+    ), f"need input file { str(ttl_dump) } for test to work"
+    in_memory: KGSource = KGSource.build(str(ttl_dump))
+    qry: str = DefaultQueryBuilder().build_sparql_query(
+        name="bodc-listing.sparql", cc="P06"
+    )
+
+    result: QueryResult = in_memory.query(sparql=qry)
+    assert result is not None, "there should be a result"
+    assert len(result) == 418, "the known dated dump had exactly 24 members"
```

Comment:
Assertion not correct.
As in the message, the len(result) is known to be 24

Reply:
have to check up - the 24 was actually ignoring a lot of matches that should be there - base on some optional part (vague memory)

I hope the same glitch is responsible for the 3->0 reduction on the 'fake' test

Reply:
query was updated to match better on the language of altLabel
and then also try avoid mulitple listings for the same member (with language variants in the altLabel)

correct result is 395